File: 0241-different-ways-to-add-parentheses/0241-different-ways-to-add-parentheses.go
func diffWaysToCompute(expression string) []int {
    var res []int
	for i := 0; i < len(expression); i++ {
		oper := expression[i]
		if oper == '+' || oper == '-' || oper == '*' {
			s1 := diffWaysToCompute(expression[:i])
			s2 := diffWaysToCompute(expression[i+1:])
			for _, a := range s1 {
				for _, b := range s2 {
					if oper == '+' {
						res = append(res, a+b)
					} else if oper == '-' {
						res = append(res, a-b)
					} else if oper == '*' {
						res = append(res, a*b)
					}
				}
			}
		}
	}

	if len(res) == 0 {
		num, _ := strconv.Atoi(expression)
		res = append(res, num)
	}

	return res
}

File: 0241-different-ways-to-add-parentheses/0241-different-ways-to-add-parentheses.rs
impl Solution {
    pub fn diff_ways_to_compute(expression: String) -> Vec<i32> {
        let mut res = Vec::new();
        for (i, oper) in expression.chars().enumerate() {
            if oper == '+' || oper == '-' || oper == '*' {
                let s1 = Solution::diff_ways_to_compute(expression[0..i].to_string());
                let s2 = Solution::diff_ways_to_compute(expression[i + 1..].to_string());
                for &a in &s1 {
                    for &b in &s2 {
                        match oper {
                            '+' => res.push(a + b),
                            '-' => res.push(a - b),
                            '*' => res.push(a * b),
                            _ => (),
                        }
                    }
                }
            }
        }
        if res.is_empty() {
            res.push(expression.parse().unwrap());
        }
        res
    }
}